circe-akka-serializer/src/main/scala/org/virtuslab/ash/CirceAkkaSerializer.scala

```diff
@@ -0,0 +1,82 @@
+package org.virtuslab.ash
+
+import akka.actor.ExtendedActorSystem
+import akka.event.{Logging, LoggingAdapter}
+import akka.serialization.SerializerWithStringManifest
+import io.circe.jawn.JawnParser
+import io.circe.{Decoder, Encoder, Printer}
+import org.reflections8.Reflections
+
+import java.io.NotSerializableException
+import java.nio.charset.StandardCharsets.UTF_8
+import java.util.NoSuchElementException
+import scala.jdk.CollectionConverters._
+import scala.reflect.runtime.{universe => ru}
+import scala.reflect.{ClassTag, classTag}
+
+abstract class CirceAkkaSerializer[Ser: ClassTag](actorSystem: ExtendedActorSystem)
+    extends SerializerWithStringManifest {
+  override val identifier = 42352
+  protected val logger: LoggingAdapter = Logging(actorSystem, this.getClass)
+
+  val codecs: Seq[(ru.TypeTag[_ <: Ser], (Encoder[_ <: Ser], Decoder[_ <: Ser]))]
+  val manifestMigrations: Seq[(String, String)]
+
+  private lazy val parents = codecs.flatMap { x =>
+    val clazz = x._1.tpe.typeSymbol.asClass
+    val mirror = ru.runtimeMirror(getClass.getClassLoader)
+    val rootClazzName = mirror.runtimeClass(clazz).getName
+    def getAllSubclasses(clazz: ru.ClassSymbol): List[ru.ClassSymbol] = {
+      if (!clazz.isSealed)
+        List(clazz)
+      else
+        clazz :: clazz.knownDirectSubclasses.toList.flatMap(x => getAllSubclasses(x.asClass))
+    }
+    getAllSubclasses(clazz).map(x => (mirror.runtimeClass(x).getName, rootClazzName))
+  }.toMap
+  private lazy val codecsMap = codecs.map(x => x.copy(_1 = x._1.tpe.typeSymbol.fullName)).toMap
+  private lazy val manifestMap = manifestMigrations.toMap
+
+  private val parser = new JawnParser
+  private val printer = Printer.noSpaces
+
+  override def manifest(o: AnyRef): String = parents.getOrElse(o.getClass.getName, "")
+
+  override def toBinary(o: AnyRef): Array[Byte] = {
+    codecsMap.get(manifest(o)) match {
+      case Some((encoder, _)) =>
+        val buffer = printer.printToByteBuffer(encoder.asInstanceOf[Encoder[AnyRef]](o), UTF_8)
+        val res = new Array[Byte](buffer.remaining)
+        buffer.get(res)
+        res
+      case None =>
+        throw new RuntimeException(
+          s"Serialization of ${o.getClass.getName} failed. Call Register[A] for this class or its supertype and append result to codecs.")
```

Comment:
New serialisation error

Reply:
Ok! for clarity, how does _appending the result to codecs_ look like in Circe? 🤔

Reply:
From tests:
```
override val codecs =
    Seq(
      Register[Tree],
      Register[StdData],
      Register[StdMigration],
      Register[TopTraitMigration],
      Register[ModifiedCodec](implicitly[ru.TypeTag[ModifiedCodec]], prepareEncoder, prepareDecoder))
```

Reply:
Ok, maybe for clarity:
```suggestion
          s"Serialization of ${o.getClass.getName} failed. Call Register[A] for this class or its supertype and append result to `def codecs`.")
```

Reply:
To make it clear that you mean `codecs`-the-symbol and not just some codecs out there